server/app/controllers/login.py

```diff
@@ -0,0 +1,109 @@
+from flask import render_template, abort, request, flash, redirect, url_for
+from app import app
+from flask_wtf import FlaskForm
+from flask_login import current_user, login_user, logout_user, UserMixin, LoginManager
+
+from wtforms import StringField, PasswordField, BooleanField, SubmitField
+from wtforms.validators import InputRequired, Email, Length
+from werkzeug.urls import url_parse
+from werkzeug.security import generate_password_hash, check_password_hash
+
+from app import app
+from app.repository import Repository, UserRole
+
+class LoginForm(FlaskForm):
+    username = StringField("Your login name", validators=[InputRequired(), Length(min=3, max=20)])
+    password = PasswordField("Your password", validators=[InputRequired(), Length(min=5, max=80)])
+    remember = BooleanField("Remember me")
+
+    submit = SubmitField("Sign In")
+
+class SatnogsUser(UserMixin):
+
+    # The following fields are useful for Flask-login
+    # It's basically a reimplementation of UserMixin
+
+    def __init__(self, user):
+        # I'm sure there's easier way to do it. There's some magic involving **...
```

Comment:
```python
for key, value in user.items():
    setattr(self, key, value)
```
But maybe ```user``` should be store as member of this class?

Reply:
But that code would only copy existing keys. It would ensure that all five fields are set. Anyway, updated as suggested.